### DISCUSSION TITLE
parseSignature on SyncBrowserHistory can go wrong

### DISCUSSION THREAD

**Discussion by francoism90** [livewire/livewire#5404] [7. Bugs (archived)] [answered]
```js
parseSignature(signature) {
        const [originalComponentId, componentName, componentIndex] = signature.split(':') // this one

        return { originalComponentId, componentName, componentIndex }
    }
```
We're registering Livewire components using the prefix `::`, e.g. for domain Invites, `app(LivewireManager::class)->component('invites::user.address', 'App\\Domain\\..')`.

This all works first on first init, but not after changing the state. The reason I think is being the `split`. It should only parse the first occurs, right?

When we change it to this: `post.component-name' => 'App\\Domain\\..`, all works fine.

**Comment by joshhanley** [comment, marked as answer]
@francoism90 yeah it's possible it was never built to support that. I would probably stick with using dot notation instead to get it to run.

Hope this helps!

**Reply by francoism90** [reply]
@joshhanley Thanks! The reason for the double dots (`::`) prefix, is to use the same logic as calling Laravel (vendor) components:
```
<x-nightshade::calendar />
<x-nightshade::color-picker />
```
It's not a problem using dot notation instead, but maybe this should be added to https://laravel-livewire.com/docs/2.x/package-dev#registering-components as a warning?

Thanks!

**Reply by joshhanley** [reply]
@francoism90 oh yeah, ideally it would be supported 😁

Yeah good idea, PR's to the docs are welcome! https://github.com/livewire/docs
